BaikalDataGenerator/Source/utils.h

```diff
@@ -0,0 +1,39 @@
+/**********************************************************************
+Copyright (c) 2018 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+********************************************************************/
+
+#pragma once
+
+#include <filesystem>
+#include <string>
+
+struct DGenConfig
+{
+    std::filesystem::path scene_file;
+    std::filesystem::path light_file;
+    std::filesystem::path camera_file;
+    std::filesystem::path spp_file;
+    std::filesystem::path output_dir;
+    int width, height;
+    bool gamma_correction;
+};
+
+#define THROW_EX(text) throw std::exception((std::string(__func__) + ": " + text).c_str());
```

Comment:
There is no such [std::exception constructor](https://en.cppreference.com/w/cpp/error/exception/exception).
Use any type from [`<stdexcept>`](https://en.cppreference.com/w/cpp/header/stdexcept) instead.